Sandbox/src/Game/Npc/NpcFactory.cpp

```diff
@@ -0,0 +1,47 @@
+#include "Game/Npc/NpcFactory.h"
+
+NpcFactory& NpcFactory::Instance()
+{
+	static NpcFactory instance;
+	return instance;
+}
+
+NpcFactory::NpcFactory()
+	: mNpcRepository(NpcRepository::Instance())
+{
+
+}
+
+NpcFactory::~NpcFactory()
+{
+
+}
+
+Npc* NpcFactory::GetNpc(int index)
```

Comment:
should we keep track of these npc pointers to delete them? or does that happen with whatever is calling `GetNpc`

Reply:
Whatever class calls ```GetNpc``` will be responsible for deleting them. This should be fine since it will likely be some type of npc handler class anyways. Although, we could always switch to using shared pointers and not have to deal with that...